src/vm/dispatchinfo.cpp

```diff
@@ -2344,8 +2335,22 @@ void DispatchInfo::MarshalParamManagedToNativeRef(DispatchMemberInfo *pMemberInf
         VARTYPE ElementVt = V_VT(pRefVar) & ~(VT_BYREF | VT_ARRAY);
         MethodTable *pElementMT = (*(BASEARRAYREF *)pSrcObj)->GetArrayElementTypeHandle().GetMethodTable();
 
+        PCODE pStructMarshalStubAddress = NULL;
+        GCPROTECT_BEGIN(*pSrcObj);
+        if ((ElementVt == VT_RECORD || ElementVt == VTHACK_NONBLITTABLERECORD) && (pElementMT != nullptr && !pElementMT->IsBlittable()))
```

Comment:
>(ElementVt == VT_RECORD || ElementVt == VTHACK_NONBLITTABLERECORD) [](start = 12, length = 66)

Shouldn't this imply the latter part of this predicate? Why would they be different?